### PR TITLE
Validate Password Reset Token On Reset Password Page

### PR DESCRIPTION
**Pull request by razamehdi-vteams**
Check Validity of The Password Reset Token when the user visits the Password Reset page.

By default, password reset is always shown regardless of whether the reset token is valid/invalid. In this pull request, i have modified the flow as follows:
- Checks Whether The Reset Token Exists in the Database.
- If token is found, then token expiration datetime is calculated. If expiration datetime is greater than the current datetime, then token is considered valid. Otherwise the token is invalid.

### CODE REVIEW

File: src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php [hunk 1]
```diff
@@ -172,6 +172,28 @@ public function createNewToken()
     }
 
     /**
+     * Get & Validate Token Data From Database.
+     *
+     * @param string $token
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php]: cs

File: src/Illuminate/Auth/Passwords/TokenRepositoryInterface.php [hunk 2]
```diff
@@ -37,4 +37,12 @@ public function delete($token);
      * @return void
      */
     public function deleteExpired();
+
+    /**
+     * Get & Validate Token Data From Database.
+     *
+     * @param string $token
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Passwords/TokenRepositoryInterface.php]: cs

File: src/Illuminate/Auth/Passwords/TokenRepositoryInterface.php [hunk 3]
```diff
@@ -37,4 +37,12 @@ public function delete($token);
      * @return void
      */
     public function deleteExpired();
+
+    /**
+     * Get & Validate Token Data From Database.
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Passwords/TokenRepositoryInterface.php]: please format this as a sentence - don't capitalize every word...

File: src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php [hunk 4]
```diff
@@ -172,6 +172,28 @@ public function createNewToken()
...
+     */
+    public function validateTokenData($token)
+    {
+        $token = $this->getTable()->where('token', $token)->first();
+
+        if (empty($token)) {
+            return false;
+        }
+
+        $expiration_date = Carbon::parse($token->created_at)->addSeconds($this->expires);
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php]: you must use camelCase

**Comment by GrahamCampbell** [comment]
Please fix all the CS issues, including those from StyleCI, and send this as one commit, not 10.

**Comment by razamehdi-vteams** [comment]
@GrahamCampbell do i need to refork the repo? If i do that, the multiple commits will again appear.

**Comment by GrahamCampbell** [comment]
> If i do that, the multiple commits will again appear.

You can just force push.

**Pull request closed without merging**